Description: Use ruby-ronn to generate documentation
 Command-line options are different. Perl regexp after generation need fixes.
Forwarded: not-needed
Author: Jérémy Lal <kapouer@melix.org>
Last-Update: 2012-04-07
--- a/scripts/doc-build.sh
+++ b/scripts/doc-build.sh
@@ -1,37 +1,5 @@
 #!/bin/bash
 
-if [[ $DEBUG != "" ]]; then
-  set -x
-fi
-set -o errexit
-set -o pipefail
-
-if ! [ -x node_modules/.bin/ronn ]; then
-  ps=0
-  if [ -f .building_ronn ]; then
-    pid=$(cat .building_ronn)
-    ps=$(ps -p $pid | grep $pid | wc -l) || true
-  fi
-
-  if [ -f .building_ronn ] && [ $ps != 0 ]; then
-    while [ -f .building_ronn ]; do
-      sleep 1
-    done
-  else
-    # a race to see which make process will be the one to install ronn
-    echo $$ > .building_ronn
-    sleep 1
-    if [ $(cat .building_ronn) == $$ ]; then
-      make node_modules/.bin/ronn
-      rm .building_ronn
-    else
-      while [ -f .building_ronn ]; do
-        sleep 1
-      done
-    fi
-  fi
-fi
-
 src=$1
 dest=$2
 name=$(basename ${src%.*})
@@ -42,7 +10,7 @@
 
 case $dest in
   *.[13])
-    ./node_modules/.bin/ronn --roff $src \
+    cat $src | ronn \
     | sed "s|@VERSION@|$version|g" \
     | perl -pi -e 's/npm\\-([^\(]*)\(1\)/npm help \1/g' \
     | perl -pi -e 's/npm\\-([^\(]*)\(3\)/npm apihelp \1/g' \
@@ -53,7 +21,7 @@
     ;;
   *.html)
     (cat html/dochead.html && \
-     ./node_modules/.bin/ronn -f $src && \
+     cat $src | ronn --fragment && \
      cat html/docfoot.html )\
     | sed "s|@NAME@|$name|g" \
     | sed "s|@DATE@|$date|g" \
@@ -62,8 +30,10 @@
     | perl -pi -e 's/npm-npm/npm/g' \
     | perl -pi -e 's/([^"-])(npm-)?README(\(1\))?/\1<a href="..\/doc\/README.html">README<\/a>/g' \
     | perl -pi -e 's/<title><a href="..\/doc\/README.html">README<\/a><\/title>/<title>README<\/title>/g' \
-    | perl -pi -e 's/([^"-])npm-([^\(]+)(\(1\))/\1<a href="..\/doc\/\2.html">\2\3<\/a>/g' \
-    | perl -pi -e 's/([^"-])npm-([^\(]+)(\(3\))/\1<a href="..\/api\/\2.html">\2\3<\/a>/g' \
+    | perl -pi -e 's/([^"-])npm-([^\(<]+)(\(1\))/\1<a href="..\/doc\/\2.html">\2\3<\/a>/g' \
+    | perl -pi -e 's/([^"-])npm-([^\(<]+)(\(3\))/\1<a href="..\/api\/\2.html">\2\3<\/a>/g' \
+    | perl -pi -e 's/([^"-])npm-([^\(<]+)(<span class="s">)(\(1\))(<\/span>)/\1<a href="..\/doc\/\2.html">\2\4<\/a>/g' \
+    | perl -pi -e 's/([^"-])npm-([^\(<]+)(<span class="s">)(\(3\))(<\/span>)/\1<a href="..\/api\/\2.html">\2\4<\/a>/g' \
     | perl -pi -e 's/([^"-])npm\(1\)/\1<a href="..\/doc\/npm.html">npm(1)<\/a>/g' \
     | perl -pi -e 's/([^"-])npm\(3\)/\1<a href="..\/api\/npm.html">npm(3)<\/a>/g' \
     | perl -pi -e 's/\([13]\)<\/a><\/h1>/<\/a><\/h1>/g' \
